File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 1
    routes = sorted(routes)
    cPosition = routes[0][1]
    for i in range(1, len(routes)):
        if routes[i][1] < cPosition:
            cPosition = routes[i][1]
        if routes[i][0] > cPosition:
            answer += 1
            cPosition = routes[i][1]
    return answer

File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    int answer = 1;
    sort(routes.begin(), routes.end());
    int cPosition=routes[0][1];
    for(int i=1;i<routes.size();i++){
     if(routes[i][1]<cPosition)
         cPosition=routes[i][1];
        if(routes[i][0]>cPosition){
            answer++;
            cPosition=routes[i][1];
        }
    }
    return answer;
}